include/range/v3/algorithm/result_types.hpp

```diff
@@ -51,6 +51,39 @@ namespace ranges
             }                                                                                   \
             /**/
 
+        #define RANGES_ALGO_RESULT_AUX_BOOL_2(C, T1, M1, T2, M2)                                \
+            constexpr explicit operator bool() const noexcept                                   \
+            {                                                                                   \
+                return value;                                                                   \
+            }                                                                                   \
```

Comment:
I would prefer not to have result structs that are implicitly convertible to their actual result. I appreciate the attempt to support the simple common use case while also returning all the computed information, but in this case, it might be best for `ends_with` to simply return `bool`.

Maybe there should be a `find_at_end` that returns all the information, and `ends_with` that just returns the `bool` result. Just spitballing here.